iib/workers/tasks/build_containerized_fbc_operations.py

```diff
@@ -0,0 +1,330 @@
+                        f"FBC fragments: {', '.join(fbc_fragments)}"
+                    ),
+                )
+
+            # Get commit SHA before waiting for the pipeline (while the temp directory still exists)
+            last_commit_sha = get_last_commit_sha(local_repo_path=local_git_repo_path)
+
+            # Wait for Konflux pipeline
+            set_request_state(request_id, 'in_progress', 'Waiting on KONFLUX build')
+
+            # find_pipelinerun has retry decorator to handle delays in pipelinerun creation
+            pipelines = find_pipelinerun(last_commit_sha)
+
+            # Get the first pipelinerun (should typically be only one)
+            pipelinerun = pipelines[0]
+            pipelinerun_name = pipelinerun.get('metadata', {}).get('name')
+            if not pipelinerun_name:
+                raise IIBError("Pipelinerun name not found in pipeline metadata")
+
+            run = wait_for_pipeline_completion(pipelinerun_name)
+
+            set_request_state(request_id, 'in_progress', 'Copying built index to IIB registry')
+            # Extract IMAGE_URL from pipelinerun results
+            image_url = get_pipelinerun_image_url(pipelinerun_name, run)
+            output_pull_specs = get_list_of_output_pullspec(request_id, build_tags)
+            # Copy the built index from Konflux to all output pull specs
+            for spec in output_pull_specs:
+                _skopeo_copy(
+                    source=f'docker://{image_url}',
+                    destination=f'docker://{spec}',
+                    copy_all=True,
+                    exc_msg=f'Failed to copy built index from Konflux to {spec}',
+                )
+                log.info("Successfully copied image to %s", spec)
+
+            # Use the first output_pull_spec as the primary one for request updates
+            output_pull_spec = output_pull_specs[0]
+            # Update request with final output
+            if not output_pull_spec:
+                raise IIBError(
+                    "output_pull_spec was not set. "
+                    "This should not happen if the pipeline completed successfully."
+                )
+
+            _update_index_image_pull_spec(
+                output_pull_spec=output_pull_spec,
+                request_id=request_id,
+                arches=arches,
+                from_index=from_index,
+                overwrite_from_index=overwrite_from_index,
+                overwrite_from_index_token=overwrite_from_index_token,
+                resolved_prebuild_from_index=from_index_resolved,
+                add_or_rm=True,
+                is_image_fbc=True,
+                # Passing an empty index_repo_map is intentional. In IIB 1.0, if
+                # the overwrite_from_index token is given, we push to git by default
+                # at the end of a request. In IIB 2.0, the commit is pushed earlier to trigger
+                # a Konflux pipelinerun. So the old workflow isn't needed.
```

Comment:
May I suggest renaming `IIB 1.0` and `IIB 2.0` into something else?

Like "legacy"/"next gen" or "traditional"/"pure Konflux"?

Because we're now in the upstream release version 9.4.3, so using `1.0` and `2.0` without context might induce interpretation error.

Reply:
good point. I named my branch `IIB 2.oh!` Maybe we can use that?